src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -34,6 +35,7 @@ public RemoteWalkContext(SourceCacheContext cacheContext, ILogger logger)
         public IList<IDependencyProvider> ProjectLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> LocalLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> RemoteLibraryProviders { get; }
+        public PackageNamespacesConfiguration PackageNamespaces { get; set; }
```

Comment:
nit: Can we do this at a later point, in a future PR?

It's not clear to me that it's necessary right now, and it'd be easier to make an argument for/against if we could see the usage.

Reply:
The reason I added this property in this PR is explained as follows.

- `RemoteWalkContext` encapsulates local/remote package sources. Hence, I added `PackageNamespacesConfiguration` for filtering packages during graph processing.
- During restore for package reference, [`RemoteDependencyWalker.WalkAsync`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs) method performs a depth first search for all the `top-level package dependencies` and their `transitive dependencies`. Hence it is required for `RemoteDependencyWalker` to be aware of `packagenamespaces`.
- My understanding is `package namespaces configuration` is also applicable to all the transitive dependency package id(s).

Please let me know if my understanding is correct. I am happy to add it in the following PR incase if that makes more sense.

Reply:
I think you have a great point, and you are very likely 100% correct.

It's just think it'd be easier to review that new addition when it's going to be used.
RestoreREquest for example is a rather obvious one, because you need the data in some restore specific model.

Reply:
Do we need `set` here? I believe most likely setting done inside constructor so no need `setter`.

Reply:
I reverted this change from this PR. I will revisit this change in my follow-up for package namespaces.